File: Practicos/1/bisection.py
def f(x):
    return -6*x**2 + 9*x + 2

def g(a, b) :
    return (a+b)/2

if __name__ == "__main__":
    a = float(input("Introduzca limite inferior: "))
    b = float(input("Introduzca limite superior: "))
    acc = float(input("Introduzca la presicion de la estimacion: ")) 

    if f(a)*f(b) < 0:
        print("En el intervalo hay al menos una raiz")

        c = g(a, b)

        if f(a)*f(c) < 0:
            b = c
        else :
            a = c

        while abs(a-b) > acc:
            c = g(a, b)
            if f(a)*f(c) < 0:
                b = c
            else :
                a = c

            print("Mid val: " + str(c))

        print("El valor es: " + str(f(c)))
    else:
        print("No hay raices")

File: Practicos/1/bisection.cpp
#include<iostream>
#include<cmath>
using namespace std;

float f(float x) {
	return -6*pow(x, 2) + 9*x + 2;
}

float g(float a, float b) {
	return (a+b)/2;
}

int main() {
	float a;
	float b;
	float acc;
	
	std::cout << "Introduzca el limite inferior ";
	std::cin >> a;
	
	std::cout << "Introduzca el limite superior ";
	std::cin >> b;
	
	std::cout << "Introduzca la presicion ";
	std::cin >> acc;
	
	if(f(a)*f(b) < 0) {
		std::cout << "En el intervalo hay al menos una raiz" << '\n';
		
		while (abs(a-b)>=acc) {
			float c = g(a, b);
			
			if(f(a)*f(c) < 0)  b = c;
			else a = c;
			
			std::cout << "Mid val: " << c << '\n';
		}
		
		std::cout << "El valor es: " << f(g(a, b)) << '\n';
	}
	else {
		std::cout << "No hay raices" << '\n';
	}
	
	return 0;
}
